libs/transloco/src/lib/transloco.signal.ts

```diff
@@ -0,0 +1,159 @@
+  : Signal<Translation>;
+
+/**
+ * Gets the translated value of a key as Signal
+ *
+ * @example
+ * text = translateSignal('hello');
+ * textList = translateSignal(['green', 'blue']);
+ * textVar = translateSignal('hello', { variable: 'world' });
+ * textSpanish = translateSignal('hello', { variable: 'world' }, 'es');
+ * textTodosScope = translateSignal('hello', { variable: 'world' }, { scope: 'todos' });
+ *
+ * @example
+ * dynamicKey = signal('hello');
+ * dynamicParam = signal({ variable: 'world' });
+ * object = translateSignal(this.dynamicKey, this.dynamicParam);
+ *
+ */
+export function translateSignal<T extends TranslateSignalKey>(
+  key: T,
+  params?: HashMap,
+  lang?: ScopeType,
+  injector?: Injector,
+): TranslateSignalRef<T> {
+  injector ??= inject(Injector);
+  const result = runInInjectionContext(injector, () => {
+    const service = inject(TranslocoService);
+    const scope = (lang ??=
+      inject(TRANSLOCO_SCOPE, { optional: true }) ?? lang);
+    return toObservable(computerKeysAndParams(key, params)).pipe(
+      switchMap((dynamic) =>
+        service.selectTranslate(
+          dynamic.key as string,
+          dynamic.params,
+          scope as string,
+        ),
+      ),
+    );
+  });
+  return toSignal(result, {
+    initialValue: Array.isArray(key) ? [] : '',
+  }) as TranslateSignalRef<T>;
+}
+
+/**
+ * Gets the translated object of a key as Signal
+ *
+ * @example
+ * object = translateObjectSignal('nested.object');
+ * title = object().title;
+ *
+ * @example
+ * dynamicKey = signal('nested.object');
+ * dynamicParam = signal({ variable: 'world' });
+ * object = translateObjectSignal(this.dynamicKey, this.dynamicParam);
+ */
+export function translateObjectSignal<T extends TranslateSignalKey>(
+  key: T,
+  params?: HashMap,
+  lang?: ScopeType,
+  injector?: Injector,
+): TranslateObjectSignalRef<T> {
+  injector ??= inject(Injector);
+  const result = runInInjectionContext(injector, () => {
+    const service = inject(TranslocoService);
+    const scope = (lang ??=
+      inject(TRANSLOCO_SCOPE, { optional: true }) ?? lang);
+    return toObservable(computerKeysAndParams(key, params)).pipe(
+      switchMap((dynamic) =>
+        service.selectTranslateObject(
+          dynamic.key as string,
+          dynamic.params,
+          scope as string,
+        ),
+      ),
+    );
+  });
+  return toSignal(result, { initialValue: Array.isArray(key) ? [] : {} });
+}
+
+function computerParams(params?: HashMap) {
+  if (!params) {
+    return signal(params).asReadonly();
+  }
+
+  if (isSignal(params)) {
+    return computed(() => params() as HashMap);
+  }
+
+  const hasSignal = Object.values(params).some(isSignal);
+  if (!hasSignal) {
+    return signal(params).asReadonly();
+  }
+
+  const getter = computed(() => {
+    return Object.entries(params).reduce((acc, [key, value]) => {
+      acc[key] = isSignal(value) ? value() : value;
+      return acc;
+    }, {} as HashMap);
+  });
+  return getter;
```

Comment:
We can just return the computed directly
```suggestion 
  return computed(() => {
    return Object.entries(params).reduce((acc, [key, value]) => {
      acc[key] = isSignal(value) ? value() : value;
      return acc;
    }, {} as HashMap);
  });
```